PowerPointLabs/PowerPointLabs/TooltipsLab/AttachTriggerAnimation.cs

```diff
@@ -0,0 +1,142 @@
+        }
+
+        private static void AddTriggerEffect(Shape triggerShape, List<Shape> shapesToAnimate, MsoAnimEffect effect, Sequence sequence)
+        {
+            // Add Entrance Effect
+            for (int i = 0; i < shapesToAnimate.Count; i++)
+            {
+                Shape animationShape = shapesToAnimate[i];
+                MsoAnimTriggerType triggerType;
+                // The first shape will be triggered by the click to appear
+                if (i == 0)
+                {
+                    triggerType = MsoAnimTriggerType.msoAnimTriggerOnShapeClick;
+                    sequence.AddTriggerEffect(animationShape, effect, triggerType, triggerShape);
+                }
+                // Rest of the shapes will appear with the first shape
+                else
+                {
+                    triggerType = MsoAnimTriggerType.msoAnimTriggerWithPrevious;
+                    sequence.AddEffect(shapesToAnimate[i], effect,
+                        MsoAnimateByLevel.msoAnimateLevelNone, MsoAnimTriggerType.msoAnimTriggerWithPrevious);
+                }
+            }
+
+            // Add Exit Effect to Shapes
+            for (int i = 0; i < shapesToAnimate.Count; i++)
+            {
+                Shape animationShape = shapesToAnimate[i];
+                MsoAnimTriggerType triggerType;
+                Effect effectInSequence;
+                // The first shape will be triggered by the click to disappear
+                if (i == 0)
+                {
+                    triggerType = MsoAnimTriggerType.msoAnimTriggerOnShapeClick;
+                    effectInSequence = sequence.AddTriggerEffect(animationShape, effect, triggerType, triggerShape);
+                }
+                // Rest of the shapes will disappear with the first shape
+                else
+                {
+                    triggerType = MsoAnimTriggerType.msoAnimTriggerWithPrevious;
+                    effectInSequence = sequence.AddEffect(shapesToAnimate[i], effect, 
+                        MsoAnimateByLevel.msoAnimateLevelNone, MsoAnimTriggerType.msoAnimTriggerWithPrevious);
+                }
+                effectInSequence.Exit = Microsoft.Office.Core.MsoTriState.msoTrue;
+            }
+        }
+
+        private static List<Shape> RemoveAnimationsInInteractiveSequence(PowerPointSlide currentSlide, Shape triggerShape, List<Shape> shapesToAnimate)
+        {
+            Sequences sequences = currentSlide.TimeLine.InteractiveSequences;
+            // A set is used here so no duplicate shapes will be added
+            ISet<Shape> shapesToAnimateSet = new HashSet<Shape>(shapesToAnimate);
+
+            // Find the existing sequence that has the triggerShape
+            for (int i = 1; i <= sequences.Count; i++)
+            {
+                Sequence sequence = sequences[i];
+                // Iterate from the back because of deletion
+                for (int j = sequence.Count; j >= 1; j--)
```

Comment:
`sequence[0] `does not need to be deleted? Is it because the index starts from 1?

Reply:
yes the index starts from 1 here

Reply:
Ok!